frontend/src/lib/k8s/pod.ts

```diff
@@ -117,6 +134,151 @@ class Pod extends makeKubeObject<KubePod>('Pod') {
 
     return stream(url, onExec, { additionalProtocols, isJson: false, ...streamOpts });
   }
+
+  private getLastRestartDate(container: KubeContainerStatus, lastRestartDate: Date): Date {
+    if (!!container.lastState?.terminated) {
+      const terminatedDate = new Date(container.lastState.terminated.finishedAt);
+      if (lastRestartDate.getTime() < terminatedDate.getTime()) {
+        return terminatedDate;
+      }
+    }
+
+    return lastRestartDate;
+  }
+
+  private hasPodReadyCondition(conditions: any): boolean {
+    for (const condition of conditions) {
+      if (condition.type === 'Ready' && condition.Status === 'True') {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // Implementation based on: https://github.com/kubernetes/kubernetes/blob/33bf8a8ebd33eae0d45c18feb1c05baa6a055cde/pkg/printers/internalversion/printers.go#L558
+  getDetailedStatus(): PodDetailedStatus {
+    // We cache this data to avoid going through all this logic when nothing has changed
+    if (
+      !!this.detailedStatusCache.details &&
+      this.detailedStatusCache.resourceVersion === this.jsonData.metadata.resourceVersion
+    ) {
+      return this.detailedStatusCache.details;
+    }
+
+    // We cache this data to avoid going through all this logic when nothing has changed
+    if (
+      !!this.detailedStatusCache.details &&
+      this.detailedStatusCache.resourceVersion === this.jsonData.metadata.resourceVersion
+    ) {
+      return this.detailedStatusCache.details;
+    }
+
+    let restarts = 0;
+    const totalContainers = (this.spec.containers ?? []).length;
+    let readyContainers = 0;
+    let message = '';
+    let lastRestartDate = new Date(0);
+
+    let reason = this.status.reason || this.status.phase;
+
+    let initializing = false;
+    for (const i in this.status.initContainerStatuses) {
+      const container = this.status.initContainerStatuses[i];
+      restarts += container.restartCount;
+      lastRestartDate = this.getLastRestartDate(container, lastRestartDate);
+
+      switch (true) {
+        case container.state.terminated?.exitCode === 0:
+          continue;
+        case !!container.state.terminated:
+          if (!container.state.terminated.reason) {
+            if (container.state.terminated.signal !== 0) {
+              reason = `Init:Signal:${container.state.terminated.signal}`;
+            } else {
+              reason = `Init:ExitCode:${container.state.terminated.exitCode}`;
+            }
+          } else {
+            reason = 'Init:' + container.state.terminated.reason;
+          }
+          message = container.state.terminated.message || '';
+          initializing = true;
+          break;
+        case !!container.state.waiting?.reason &&
+          container.state.waiting.reason !== 'PodInitializing':
+          reason = 'Init:' + container.state.waiting.reason;
+          initializing = true;
+          message = container.state.waiting.message || '';
```

Comment:
This message part isn't in the golang function.

Reply:
Right. This is an implementation based on the golang one.
Is this something problematic in your opinion? (Any alternatives if so?)